File: backend/src/Tmap.Api/Features/NoteGroups/NoteGroupDtos.cs
namespace Tmap.Api.Features.NoteGroups;

public sealed record CreateNoteGroupRequest(
    string Name,
    string? Emoji,
    Guid? ProjectId,
    string? Rank = null,
    Guid? Id = null);

public sealed record UpdateNoteGroupRequest(
    string? Name,
    string? Emoji,
    Guid? ProjectId,
    string? Rank);

public sealed record NoteGroupResponse(
    Guid Id,
    string Name,
    string Emoji,
    Guid? ProjectId,
    string Rank,
    DateTimeOffset CreatedAt,
    DateTimeOffset UpdatedAt);

public sealed record ReorderItem(Guid Id, string Rank);

File: android/app/src/main/java/net/qmindtech/tmap/data/remote/dto/NoteGroupDtos.kt
package net.qmindtech.tmap.data.remote.dto

import kotlinx.serialization.Serializable

@Serializable
data class CreateNoteGroupRequest(
    val id: String? = null,
    val name: String,
    val emoji: String,
    val projectId: String? = null,
    val rank: String? = null,
)

@Serializable
data class UpdateNoteGroupRequest(
    val name: String? = null,
    val emoji: String? = null,
    val projectId: String? = null,
    val rank: String? = null,
)

@Serializable
data class NoteGroupResponse(
    val id: String,
    val name: String,
    val emoji: String,
    val projectId: String?,
    val rank: String?,
    val createdAt: String,
    val updatedAt: String,
)
